swadm/src/compliance.rs

```diff
@@ -0,0 +1,613 @@
+    all: bool,
+) -> anyhow::Result<()> {
+    // Get all switch ports
+    let all_ports = client
+        .port_list()
+        .await
+        .context("failed to list switch ports")?
+        .into_inner();
+
+    // Filter switch ports based on --all flag and pattern
+    let target_ports: Vec<types::PortId> = if all {
+        // If --all is specified, apply pattern filtering to all ports
+        if let Some(user_pattern) = pattern {
+            all_ports
+                .into_iter()
+                .filter(|port| port.to_string().contains(user_pattern))
+                .collect()
+        } else {
+            all_ports
+        }
+    } else {
+        // If --all is not specified, filter to qsfp ports then apply pattern
+        let qsfp_ports: Vec<types::PortId> = all_ports
+            .into_iter()
+            .filter(|port| matches!(port, types::PortId::Qsfp(_)))
+            .collect();
+
+        if let Some(user_pattern) = pattern {
+            qsfp_ports
+                .into_iter()
+                .filter(|port| port.to_string().contains(user_pattern))
+                .collect()
+        } else {
+            qsfp_ports
+        }
+    };
+
+    let mut tw = TabWriter::new(stdout());
+    writeln!(
+        tw,
+        "{}\t{}\t{}\t{}\t{}",
+        "PORT".underline(),
+        "POWER".underline(),
+        "LINK".underline(),
+        "LINK-ENABLED?".underline(),
+        "LINK-STATE".underline()
+    )?;
+
+    // For each port, check for links and power state
+    for port_id in target_ports {
+        // Get power state for the port
+        let power_state = match client.transceiver_power_get(&port_id).await {
+            Ok(power) => {
+                let state = power.into_inner();
+                // Convert to a cleaner string representation
+                format!("{:?}", state).to_lowercase()
+            }
+            Err(_) => "N/A".to_string(),
+        };
+
+        let links = client
+            .link_list(&port_id)
+            .await
+            .context(format!("failed to list links for port {}", port_id))?
+            .into_inner();
+
+        if links.is_empty() {
+            // Port has no links
+            writeln!(tw, "{}\t{}\t-\t-\tNo links", port_id, power_state)?;
+        } else {
+            // Port has links - show each link with the same power state
+            for link in links {
+                writeln!(
+                    tw,
+                    "{}\t{}\t{}\t{}\t{}",
+                    link.port_id,
+                    power_state,
+                    link,
+                    link.enabled,
+                    link.link_state
+                )?;
+            }
+        }
+    }
+
+    tw.flush().map_err(|e| e.into())
+}
+
+async fn compliance_ports_enable(
+    client: &Client,
+    pattern: Option<&str>,
+    enabled: bool,
+    all: bool,
+) -> anyhow::Result<()> {
+    let links = get_matching_links(client, pattern, all).await?;
+
+    for link in links {
```

Comment:
Not a big deal, but I think this could be tightened up a bit:
```
    let op = if enabled { "enable" } else { "disable" };
    for link in links {
        client
            .link_enabled_set(&link.port_id, &link.link_id, enabled)
            .await
            .context(format!("failed to {op} link {link}",))?;
        println!("{op}d link {link}");
    }